sdk/ai/Azure.AI.Projects/src/Custom/Agent/AgentsClient.cs

```diff
@@ -171,18 +172,57 @@ public virtual Task<Response<PageableList<RunStep>>> GetRunStepsAsync(
         {
             Argument.AssertNotNull(run, nameof(run));
-            return SubmitToolOutputsToRun(run.ThreadId, run.Id, toolOutputs, null, cancellationToken);
+            Response response = SubmitToolOutputsToRunInternal(run.ThreadId, run.Id, toolOutputs, false, cancellationToken);
+            return Response.FromValue(ThreadRun.FromResponse(response), response);
         }
 
         /// <summary> Submits outputs from tool calls as requested by a run with a status of 'requires_action' with required_action.type of 'submit_tool_outputs'. </summary>
         /// <param name="run"> The <see cref="ThreadRun"/> that the tool outputs should be submitted to. </param>
         /// <param name="toolOutputs"> The list of tool call outputs to provide as part of an output submission to an agent thread run. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="run"/>  is null. </exception>
-        public virtual Task<Response<ThreadRun>> SubmitToolOutputsToRunAsync(ThreadRun run, IEnumerable<ToolOutput> toolOutputs, CancellationToken cancellationToken = default)
+        public async virtual Task<Response<ThreadRun>> SubmitToolOutputsToRunAsync(ThreadRun run, IEnumerable<ToolOutput> toolOutputs, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNull(run, nameof(run));
-            return SubmitToolOutputsToRunAsync(run.ThreadId, run.Id, toolOutputs, null, cancellationToken);
+            Response response = await SubmitToolOutputsToRunInternalAsync(run.ThreadId, run.Id, toolOutputs, false, cancellationToken).ConfigureAwait(false);
+            return Response.FromValue(ThreadRun.FromResponse(response), response);
+        }
+
+        /// <summary> Submits outputs from tools as requested by tool calls in a run. Runs that need submitted tool outputs will have a status of 'requires_action' with a required_action.type of 'submit_tool_outputs'. </summary>
+        /// <param name="threadId"> Identifier of the thread. </param>
+        /// <param name="runId"> Identifier of the run. </param>
+        /// <param name="toolOutputs"> A list of tools for which the outputs are being submitted. </param>
+        /// <param name="stream"> If true, returns a stream of events that happen during the Run as server-sent events, terminating when the run enters a terminal state. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="threadId"/>, <paramref name="runId"/> or <paramref name="toolOutputs"/> is null. </exception>
+        private Response SubmitToolOutputsToRunInternal(string threadId, string runId, IEnumerable<ToolOutput> toolOutputs, bool stream, CancellationToken cancellationToken)
```

Comment:
Why do we need this internal method? Is there anything we need that is not available in the generated method [here](https://github.com/Azure/azure-sdk-for-net/blob/3575d8d3466cf825fafe38eaea5f11cc287ad3e6/sdk/ai/Azure.AI.Projects/src/Generated/AgentsClient.cs#L2282)?

Reply:
The problem is that generated method is trying to deserialize the thread:
```
Response.FromValue(ThreadRun.FromResponse(response), response);
```
that results in the error as we need to deserialize to streaming response.

Reply:
The method which are calling SubmitToolOutputsToRunInternal are not streaming, you are passing streaming false - https://github.com/Azure/azure-sdk-for-net/blob/72baa09c69898e1df108a8bbbbb3f6dd8b707acc/sdk/ai/Azure.AI.Projects/src/Custom/Agent/AgentsClient.cs#L184

Reply:
Fixed.